### PR TITLE
Upgrade jupyterbook

### LINKED ISSUE #692: Jupyter Book update breaks CI
### Summary

Jupyter Book has a new major release (0.6.0): https://github.com/jupyter/jupyter-book/releases/tag/v0.6.0

And it's SUPER LOVELY (I really like the download as PDF button and the footer) but it breaks our CI.

So @KirstieJane & @alexmorley have implemented a hot fix in #691 to pin to the old version of Jupyter Book....but fixing what's going wrong so we can use the 0.6 version would be better!

### What needs to be done?

- [ ] Figure out why the build is failing with the old jupyter book
- [ ] Fix that, and then fix the version of jupyter book to 0.6 in `requirements.txt`

### Who can help?

* Anyone who knows about jekyll, netlify, jupyter book and some combination of those tools. @alexmorley @emdupre and @choldgraf are already in a different thread 💖 

---

### Updates

*No updates so far*

### PR DESCRIPTION
**Pull request by KirstieJane**
### Summary

Fixes #692 

We're way behind jupyter book at the moment, this WIP PR tries to fix that!

### List of changes proposed in this PR (pull-request)

* Update requirements from v 0.5.2 to 0.6.4
* *Lorem ipsum dolor sit amet, consectetur adipiscing.*

### What should a reviewer concentrate their feedback on?

- [ ] *Lorem ipsum dolor sit amet, consectetur adipiscing.*
- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by fdiblen** [comment]
Temporary workaround was to install an old version of the jupyter-book:

```bash
pip install jupyter-book==0.4.2
```

**Comment by martinagvilas** [comment]
To prevent the CI from failing I think we need to:

1. Add to the .travis.yml (I can do this in a separate PR)
```
    - language: python
      python: 3.6	      
```

2. Merge master into this branch so that the bad-latin errors fixed by @sgibson91 get updated here as well.

**Comment by martinagvilas** [comment]
I also think we need to update the [requirements.txt](https://github.com/alan-turing-institute/the-turing-way/blob/master/book/website/requirements.txt) to a newer version of jupyter-book

File: book/website/requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ numpy
 datascience
 folium
 matplotlib
-jupyter-book==0.5.2
+jupyter-book
```
Comment (sgibson91) [inline book/website/requirements.txt]: 0.6.4 is the version I used to update the book, so we should pin it to that until we feel confident in making future upgrades

**Comment by martinagvilas** [comment]
> Is it possible to define two languages in travis? (I'm learning so many cool things from this PR!)

I don't think so, I was thinking of replacing python by generic but I can see now why to use generic (I totally ignored that commented line). Then the only way of upgrading python and keeping the generic language would be to manually install python 3.7 then?

> We definitely need Python 3.6+, but I'd personally go with 3.7. 3.8 is now available and I think staying 1 minor version behind for stability is good practice.

Makes sense!

File: book/website/requirements.txt (hunk 1 above)
Comment (martinagvilas) [reply book/website/requirements.txt]: great, I can pin it to that version

**Comment by sgibson91** [comment]
> Then the only way of upgrading python and keeping the generic language would be to manually install python 3.7 then?

Yeah, manual install of 3.7 would be the way to go.

These lines say that it's getting a recent version of python, but I can only really see the pip install:
https://github.com/alan-turing-institute/the-turing-way/blob/a772210867ee96fbf57941721edd9b25a69cd079/.travis.yml#L6-L9

I'd try putting these lines at the top of the `before_install` block and see if that helps.

```
sudo apt-get update
sudo apt-get install python3.7
```

**Comment by martinagvilas** [comment]
Travis still tries to use python 3.5 after the 3.7 installation

I pushed a possible fix for this [here](https://github.com/alan-turing-institute/the-turing-way/commit/a59d2b0a2a6e3938591e390756f9057ff56d2057) but for some reason the commit shows in the branch but not in this PR.

**Comment by martinagvilas** [comment]
Finally! I had to remove the `bash ./scripts/html-proof.sh ` line from `.travis `since that script was removed from the repository in this PR, so I assumed that check was no longer necessary. Let me know if instead we should add back the script and add the check to the ci again.

https://deploy-preview-844--the-turing-way.netlify.com seems to show that the upgrade now looks fine.

**Comment by sgibson91** [comment]
Well done @martinagvilas! Way to persevere! I think that script is required though. html-proofer checks for broken links and other html elements, so it's quite important. The script should be in `book/website/scripts/`

**Comment by martinagvilas** [comment]
Okey, I will fetch the script from master and uncomment the line in the ci so we can test for it :)

**Comment by martinagvilas** [comment]
I think we got it 🚀

**Comment by KirstieJane** [comment]
OMG Look at those green ticks!! @sgibson91 @martinagvilas - are we good to merge! Click the green button if so! 🚀 🌟

**Comment by sgibson91** [comment]
I was letting @martinagvilas do the honours since she worked so hard 🎉

**Comment by martinagvilas** [comment]
Merged 😄  We can optimize new features like the footers and the headers on another PR 🚀

**Comment by KirstieJane** [comment]
AAAAAAAAAMAAAAAZING! Thank you so much @martinagvilas 💖

**Pull request merged**